file-manager/fileManager.cpp

```diff
@@ -0,0 +1,168 @@
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/json.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/uri.hpp>
+
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <filesystem>
+
+// Define keys for the key-value pair.
+#define FILE_NAME "name"
+#define FILE_PATH "path"
+#define FILE_DATA "data"
+
+// ********************************************** Helper Functions **********************************************
+
+// Upload a file to the collection.
+bool upload(const std::string& filePath, mongocxx::collection& collection)
+{
+    // Open the binary file
+    std::ifstream file(filePath, std::ios::binary | std::ios::ate);
+    if (!file) 
+    {
+        std::cout << "Failed to open the file: " << filePath << std::endl;
+        return false;
+    }
+
+    // Get the file size.
+    std::streamsize fileSize = file.tellg();
+    file.seekg(0, std::ios::beg);
+
+    // Read the file content into a buffer
+    std::vector<char> buffer(fileSize);
+    if (!file.read(buffer.data(), fileSize)) 
+    {
+        std::cout << "Failed to read the file: " << filePath << std::endl;
+        return false;
+    }
+
+    // Create the binary object for bsoncxx.
+    bsoncxx::types::b_binary data{bsoncxx::binary_sub_type::k_binary, static_cast<std::uint32_t>(fileSize), reinterpret_cast<const std::uint8_t*>(buffer.data())};
+
+    // Create a document with the file name and file content.
+    auto doc = bsoncxx::builder::stream::document{};
+    doc << FILE_NAME << std::filesystem::path(filePath).filename();
+    doc << FILE_PATH << filePath;
+    doc << FILE_DATA << data;
```

Comment:
```suggestion
    auto doc = make_document(
        kvp(FILE_NAME, std::filesystem::path(filePath).filename()),
        kvp(FILE_PATH, filePath),
        kvp(FILE_DATA, data));
```

And add needed `using` statements above the function:
```c++
    using bsoncxx::builder::basic::kvp;
    using bsoncxx::builder::basic::make_document;
```

Use of the stream builder is discouraged. See https://mongocxx.org/mongocxx-v3/working-with-bson/:

> Because doing this properly is difficult and the compiler error messages can be confusing, using the stream builder is discouraged.